utils/multires/generate.py

```diff
@@ -42,59 +43,107 @@
+parser.add_argument('-B', '--avoidbackground', action='store_true',
+                    help='viewer should limit view to avoid showing background, so using --backgroundcolor is not needed')
+parser.add_argument('-a', '--autoload', action='store_true',
+                    help='automatically load panorama in viewer')
 parser.add_argument('-q', '--quality', dest='quality', default=75, type=int,
                     help='output JPEG quality 0-100')
 parser.add_argument('--png', action='store_true',
                     help='output PNG tiles instead of JPEG tiles')
 parser.add_argument('-n', '--nona', default=nona, required=nona is None,
                     metavar='EXECUTABLE',
                     help='location of the nona executable to use')
+parser.add_argument('-G', '--gpu', action='store_true',
+                    help='perform image remapping by nona on the GPU')
+parser.add_argument('-d', '--debug', action='store_true',
+                    help='debug mode (print status info and keep intermediate files)')
 args = parser.parse_args()
 
 # Process input image information
 print('Processing input image information...')
 origWidth, origHeight = Image.open(args.inputFile).size
-if float(origWidth) / origHeight != 2:
-    print('Error: the image width is not twice the image height.')
-    print('Input image must be a full, not partial, equirectangular panorama!')
-    sys.exit(1)
+haov = args.haov
+if haov == -1:
+    if args.cylindrical or float(origWidth) / origHeight == 2:
+        print('Assuming --haov 360.0')
+        haov = 360.0
+    else:
+        print('Unless given the --haov option, equirectangular input image must be a full (not partial) panorama!')
+        sys.exit(1)
+vaov = args.vaov
+if vaov == -1:
+    if args.cylindrical or float(origWidth) / origHeight == 2:
+        print('Assuming --vaov 180.0')
+        vaov = 180.0
+    else:
+        print('Unless given the --vaov option, equirectangular input image must be a full (not partial) panorama!')
+        sys.exit(1)
 if args.cubeSize != 0:
     cubeSize = args.cubeSize
 else:
     cubeSize = 8 * int(origWidth / math.pi / 8)
-levels = int(math.ceil(math.log(float(cubeSize) / args.tileSize, 2))) + 1
+tileSize = min(args.tileSize, cubeSize)
+levels = int(math.ceil(math.log(float(cubeSize) / tileSize, 2))) + 1
 origHeight = str(origHeight)
 origWidth = str(origWidth)
 origFilename = os.path.join(os.getcwd(), args.inputFile)
 extension = '.jpg'
 if args.png:
     extension = '.png'
+colorList = eval(args.backgroundColor)
+colorTuple = (int(colorList[0]*255), int(colorList[1]*255), int(colorList[2]*255))
+
+if args.debug:
+    print('maxLevel: '+ str(levels))
+    print('tileResolution: '+ str(tileSize))
+    print('cubeResolution: '+ str(cubeSize))
 
 # Create output directory
-os.makedirs(args.output)
+if not os.path.exists(args.output):
+    os.makedirs(args.output)
```

Comment:
This was intended to fail if the output directory already exists. A check could be added to print a descriptive error message instead of a stack trace, but I don't think the script should continue if the output directory already exists.